playbooks/robusta_playbooks/prometheus_enrichments.py

```diff
@@ -42,6 +47,52 @@ def parse_duration(
     return None, None
 
 
+@action
+def prometheus_all_available_metrics(event: ExecutionBaseEvent, prometheus_params: PrometheusParams):
+    result = get_prometheus_all_available_metrics(prometheus_params=prometheus_params)
+
+    event.add_enrichment([JsonBlock(json.dumps({"metrics": result}))])
+
+
+def get_prometheus_all_available_metrics(prometheus_params: PrometheusParams) -> List[str]:
+    try:
+        prom = get_prometheus_connect(prometheus_params=prometheus_params)
+        return prom.all_metrics()
+
+    except Exception as e:
+        raise Exception(f"An error occurred while fetching all available Prometheus metrics: {e}")
+
+
+class PrometheusGetSeriesParams(PrometheusParams):
+    """
+    :var match: List of Prometheus series selectors.
+    :var start_time: Optional start time for the query as datetime.
+    :var end_time: Optional end time for the query as datetime.
+    """
+
+    match: List[str]
+    start_time: Optional[datetime] = None
+    end_time: Optional[datetime] = None
+
+
+@action
+def prometheus_get_series(event: ExecutionBaseEvent, prometheus_params: PrometheusGetSeriesParams):
+    result = get_prometheus_series(prometheus_params=prometheus_params)
+
+    event.add_enrichment([JsonBlock(json.dumps({"series": result}))])
+
+
+def get_prometheus_series(prometheus_params: PrometheusGetSeriesParams) -> dict:
+    try:
+        prom = get_prometheus_connect(prometheus_params=prometheus_params)
+        return prom.get_series(match=prometheus_params.match, end_time=prometheus_params.end_time,
+                               start_time=prometheus_params.start_time)
+
+    except Exception as e:
+        raise Exception(f"Failed to fetch Prometheus series for match criteria {prometheus_params.match} "
```

Comment:
If you're raising an error this way, you lose the original stack trace
I suggest logging the error, with exc_info=true, and raising the original exception
```
logging.error(f"Failed to fetch Prometheus series for match criteria {prometheus_params.match} within the time range {prometheus_params.start_time} - {prometheus_params.end_time}", exc_info=True)
raise e
```